compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -312,6 +312,11 @@ impl<'tcx> SizeSkeleton<'tcx> {
     ) -> Result<SizeSkeleton<'tcx>, &'tcx LayoutError<'tcx>> {
         debug_assert!(!ty.has_non_region_infer());
 
+        let ty = match *ty.kind() {
+            ty::Pat(inner_ty, pat) if matches!(*pat, ty::PatternKind::Range { .. }) => inner_ty,
```

Comment:
Why is this needed?

Reply:
It is not for this PR, I think this was needed to make newindex use pattern types or some of the other conversion I was doing from rustc_scalar attributes to pattern types but this changed ended sneaking into this PR.